shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/connection/BackendConnection.java

```diff
@@ -81,7 +82,7 @@
     
     private final Collection<DatabaseCommunicationEngine> inUseDatabaseCommunicationEngines = Collections.newSetFromMap(new ConcurrentHashMap<>(64));
     
-    private final Collection<ConnectionPostProcessor> connectionPostProcessors = new LinkedList<>();
+    private final Collection<ConnectionPostProcessor> connectionPostProcessors = new CopyOnWriteArrayList<>();
```

Comment:
Could we fix this issue by defer the flush only? CopyOnWriteArrayList may impact the performance.

Reply:
connectionPostProcessors has memory visibility issues here,linkedList is not thread-safe,if we don`t use CopyOnWriteArrayList, we can use Collections.synchronizedList(new LinkedList<>()) instead.
Do you have any recommendations？

Reply:
I took a look at the api doc,Collections.synchronizedList(new LinkedList()) need use synchronized (list) when Iterate over it.
so if we want to keep use LinkedList,we need to synchronized (connectionPostProcessors /backendConnection) when add item or foreach it.

Reply:
My understanding is if we defer the flush, client won't send next packet until the resources are closed and the processors will be operated by only one thread.

Reply:
As you said, processors are indeed only operated in one thread at a time, but not promise in the same thread. The changes of thread1 may not be visible to thread2, which is why we use volatile.I`ll test it